btsieve/src/bitcoin/bitcoind_http_blocksource.rs

```diff
@@ -0,0 +1,205 @@
+use crate::blocksource::{self, BlockSource};
+use bitcoin_support::{deserialize, Block, Network, Transaction};
+use futures::{Future, Stream};
+use reqwest::r#async::Client;
+use serde::Deserialize;
+use std::time::Duration;
+use tokio::timer::Interval;
+
+#[derive(Deserialize)]
+struct ChainInfo {
+    bestblockhash: String,
+}
+
+#[derive(Debug)]
+pub enum Error {
+    Reqwest(reqwest::Error),
+    Hex(hex::FromHexError),
+    BlockDeserialization(bitcoin_support::consensus::encode::Error),
+    TransactionDeserialization(String),
+}
+
+#[derive(Clone)]
+pub struct BitcoindHttpBlockSource {
+    network: Network,
+    base_url: String,
+    client: Client,
+}
+
+impl BitcoindHttpBlockSource {
+    pub fn new(url: String, network: Network) -> Self {
+        Self {
+            network,
+            base_url: url,
+            client: Client::new(),
+        }
+    }
+
+    fn latest_block(&self) -> impl Future<Item = Block, Error = Error> + Send + 'static {
+        let cloned_self = self.clone();
+
+        self.latest_block_hash()
+            .and_then(move |latest_block_hash| cloned_self.block_by_hash(latest_block_hash))
+    }
+
+    fn latest_block_hash(&self) -> impl Future<Item = String, Error = Error> + Send + 'static {
+        let bitcoind_blockchain_info_url = format!("{}/rest/chaininfo.json", self.base_url);
+
+        self.client
+            .get(bitcoind_blockchain_info_url.as_str())
+            .send()
+            .map_err(|e| {
+                log::error!("Error when sending request to bitcoin-d");
+                Error::Reqwest(e)
+            })
+            .and_then(move |mut response| {
+                response.json::<ChainInfo>().map_err(|e| {
+                    log::error!("Error when deserialising the response from bitcoin-d");
+                    Error::Reqwest(e)
+                })
+            })
+            .map(move |blockchain_info| blockchain_info.bestblockhash)
+    }
+
+    fn block_by_hash(
+        &self,
+        block_hash: String,
+    ) -> impl Future<Item = Block, Error = Error> + Send + 'static {
+        let raw_block_by_hash_url = format!("{}/rest/block/{}.hex", self.base_url, block_hash);
+
+        self.client
+            .get(raw_block_by_hash_url.as_str())
+            .send()
+            .map_err(Error::Reqwest)
+            .and_then(|mut response| response.text().map_err(Error::Reqwest))
+            .and_then(move |response_text| {
+                let mut response_text: String = response_text;
+                remove_tailing_linebreak(&mut response_text);
+                hex::decode(response_text).map_err(Error::Hex)
+            })
+            .and_then(|bytes| deserialize(bytes.as_ref()).map_err(Error::BlockDeserialization))
+            .map(move |block| {
+                log::trace!("Got {:?}", block);
+                block
+            })
+    }
+
+    /// Currently needed because we don't store the complete transaction in the
```

Comment:
I am not sure these comments are necessary.
I'd be keen to drop them and have an issue open to store and return complete transaction instead.